### PR TITLE
Make Ratelimit TLS Certs optional

### PR DESCRIPTION
**Pull request by arkodg**
* Certs are not required for simple TLS

### CODE REVIEW

**Comment by arkodg** [comment]
cc @qicz

File: internal/infrastructure/kubernetes/ratelimit/resource.go [hunk 1]
```diff
@@ -185,20 +185,23 @@ func expectedRateLimitContainerEnv(rateLimit *egcfgv1a1.RateLimit, rateLimitDepl
...
+
+		if rateLimit.Backend.Redis.TLS.CertificateRef != nil {
+			env = append(env, []corev1.EnvVar{
+				{
+					Name:  RedisTLSClientCertEnvVar,
+					Value: RedisTLSClientCertFilename,
+				},
+				{
+					Name:  RedisTLSClientKeyEnvVar,
+					Value: RedisTLSClientKeyFilename,
```
Comment (qicz) [inline internal/infrastructure/kubernetes/ratelimit/resource.go:201]: needs ca.crt ? 



https://github.com/envoyproxy/gateway/blob/a7746d648bde075a7b841bc576103ec16a8ee08e/internal/infrastructure/kubernetes/ratelimit/resource.go#L195-L218

Comment (arkodg) [reply internal/infrastructure/kubernetes/ratelimit/resource.go:201]: raised a separate issue to track Ca cert
https://github.com/envoyproxy/gateway/issues/1419

Comment (arkodg) [reply internal/infrastructure/kubernetes/ratelimit/resource.go:201]: oh nvm, we should delete the CACertEnvVar for now

Comment (qicz) [reply internal/infrastructure/kubernetes/ratelimit/resource.go:201]: this env just exists in past version and removed ref your comment

**Review by qicz — APPROVED** [review]
LGTM

**Review by zirain — APPROVED** [review]

**Pull request merged**